### PR TITLE
build: better sharing of material test utilities

### PR DESCRIPTION
**Pull request by DevVersion**
* The slide toggle tests no longer import the test gesture config from `../slider/`
* The datepicker month constants from `@angular/material/core` have been moved to `lib/testing`. The old export from `core` will be kept for backwards compatibility until V8

### CODE REVIEW

File: src/lib/testing/BUILD.bazel [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package(default_visibility=["//visibility:public"])
+
+load("//tools:defaults.bzl", "ng_module")
+
+ng_module(
+  name = "testing",
```
Comment (jelbourn) [inline src/lib/testing/BUILD.bazel]: We should add another item to #13456 to ensure that the release doesn't include `cdk/testing` or `material/testing`

File: src/lib/testing/README.md [hunk 2]
```diff
@@ -0,0 +1,2 @@
+Note that this is not an entry-point. This directory is just used to share testing logic
+that is specific to Angular Material.
```
Comment (jelbourn) [inline src/lib/testing/README.md]: I would reword this a bit:
```markdown
The code in this `testing/` directory is not published to npm and is not a public API; it is
only used for internal testing of Angular Material components.
```

File: src/lib/testing/BUILD.bazel (hunk 1 above)
Comment (DevVersion) [reply src/lib/testing/BUILD.bazel]: Good idea. I will add it to the issue

**Comment by DevVersion** [comment]
@jelbourn Re-added the constants in favor of backwards compatibility. Please have another look.

**Comment by andrewseguin** [comment]
Sorry, can you rebase again?

**Comment by DevVersion** [comment]
@andrewseguin This one is rebased now. It's just red because of the API guardian tests. Those should be fixed with #14144.
